kernel/kalloc.c

```diff
@@ -21,12 +21,14 @@ struct run {
 struct {
   struct spinlock lock;
   struct run *freelist;
+  uint64 freelist_sz;
```

Comment:
I don't want to chase pointers every time we need to get the free size, so I added a counter to the kmem struct, which should always be up to date due to its adjustment in kfree and kalloc.